include/libdbc/message.hpp

```diff
@@ -3,21 +3,57 @@
 
 #include <string>
 #include <vector>
+#include <array>
 #include <iostream>
 #include <libdbc/signal.hpp>
 
 namespace libdbc {
 	struct Message {
-		uint32_t id;
-		std::string name;
-		uint8_t size;
-		std::string node;
-		std::vector<Signal> signals;
-
 		Message() = delete;
 		explicit Message(uint32_t id, const std::string& name, uint8_t size, const std::string& node);
 
-		virtual bool operator==(const Message& rhs) const;
+        /*!
+         * \brief parseSignals
+         * \param data
+         * \param values
+         * \return
+         */
+        bool parseSignals(const std::vector<uint8_t>& data, std::vector<double> &values) const;
+        bool parseSignals(const std::array<uint8_t,8>& data, std::vector<double>& values) const;
+        bool parseSignals(const uint8_t* data, std::vector<double>& values) const;
+
+        void appendSignal(const Signal& signal);
+        const std::vector<Signal> signals() const;
+        uint32_t id() const;
+
+        /*!
+         * \brief prepareMessage
+         * Preparing message to be able to parse signals afterwards. This speeds up parsing
+         */
+        void prepareMessage();

```

Comment:
How does sorting them speed up parsing?

Reply:
in the current approach not. I tried to reimplement the parser like in the python CAN tools, but I was not yet finished. I think the approach I currently have for <8Byte length and little endian is really fast, only 2 shifts + 3 operations (+-).